### PR TITLE
Profile badges

### PR DESCRIPTION
**Pull request by colinta**
Pending: all the 'learn more' links, staging server w/ badges for API testing.

- if any badges are visible, the "total views" area is split to accomodate the new view
- tapping a badge opens a modal w/ "learn more"
- tapping "+N" or anywhere else on the badges view opens the "see all badges" list
- tapping "Learn more" opens a web browser
- if 'badges' is not returned we check 'user.categories' as a fall-back.

[Finishes #143609479](https://www.pivotaltracker.com/story/show/143609479)

### CODE REVIEW

**Comment by ellodanger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

**Comment by steam** [comment]
@colinta, I believe that this isn't yet ready to review because of the WIP in the PR title but I don't want to hold you up if it is. Let me know when you want a review. 🤘

**Comment by colinta** [comment]
@steam this is GTG/ready for review.  The only feature not present is the fetching of `badges.json`.  I checked w/ casey and lucian, they're fine having that in the next version, freeing me up to work on editorialized homepage.  So this version will ignore any badges not defined in `ProfileBadge`.

**Comment by colinta** [comment]
Rebasing to include all `release/2.3` code/spec fixes.

**Comment by colinta** [comment]
```
Test Suite 'All tests' passed at 2017-05-01 12:54:02.309.
	 Executed 3838 tests, with 0 failures (0 unexpected) in 407.937 (409.509) seconds
```

File: Sources/Model/ProfileBadge.swift [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+            return InterfaceString.Badges.StaffLink
+        default:
+            return InterfaceString.Badges.LearnMore
+        }
+    }
+
+    var url: URL? {
+        switch self {
+        case .staff:
+            return URL(string: "https://ello.co/wtf/about/team/")
```
Comment (colinta) [inline Sources/Model/ProfileBadge.swift]: This was temporary code anyway, the `badges.json` refactor gets rid of the links, and uses what's in the json file instead.

**Pull request merged**